### ISSUE TITLE
Bump supported version

### ISSUE THREAD

**Issue by ionut-arm** [parallaxsecond/rust-tss-esapi#207]
We're currently targeting version 2.3.3 of the library, with versions 2.4.0 and up probably having an inconsistent API with our abstractions.

There are a couple of options for us to move forward - we can either drop support for 2.3.3 completely and simply continue with 2.4.0, or we can try and keep them both alive as features.

**Comment by hug-dev** [comment]
Also to keep in mind that RHEL uses `2.3.2`.

**Comment by ionut-arm** [comment]
I've tried it locally - just pulling the current tip of `master` of this repo and running the `all-ubuntu.sh` with various versions installed, 2.3.2, 2.3.3, 2.4.0 all seem to pass all tests. @Superhepper - was there some specific use case that failed for you for 2.4.0? Maybe I'm testing the wrong thing.

For Parsec I've tried all those versions as well (though with the latest release, not with the current state). 2.3.2 clearly fails to work (because of a bug that was fixed in 2.3.3). 2.4.0 does work. I'll try using the current tip, see if it still works.

**Comment by Superhepper** [comment]
@ionut-arm 
Nah, I changed the code previously in order to make it work will all versions. So instead of using the correct interface type when converting a native interface type to a TPM interface type ( because the correct TSS interface type had the wrong name) I used some kind of 'base type' and as I said previously it is not something critical. Because so far I have not found something so fundamentally broken that it cannot be fixed with a little bit of a hack. But it would be nice to be prepare.

**Comment by Superhepper** [comment]
> 
> 
> Also to keep in mind that RHEL uses `2.3.2`.

I know it was a real pain that we support 2.3.3 and RHEL supports 2.3.2 that is something that I have had to deal with.

**Comment by ionut-arm** [comment]
Ok, we can keep this open for now with no pending action just yet

**Comment by Superhepper** [comment]
In main the minim tpm2-tss version has been set to 4.1.3 so I think this can be closed.
